lib/bypass/instance.ex

```diff
@@ -81,7 +81,7 @@ defmodule Bypass.Instance do
   end
   defp do_handle_call(:down, from, %{socket: socket, ref: ref} = state) when not is_nil(socket) do
     nil = state.caller_awaiting_down  # assertion
-    if state.retained_plug != nil do
+    if state.retained_plugs != [] do
```

Comment:
This waits for all retained plugs to finish instead of the specific plug associated with this Bypass instance.